File: network/src/main/kotlin/org/bitlap/network/BSQLException.kt
package org.bitlap.network

import java.sql.SQLException

/**
 * driver exception
 * please use this when we do not need error stack.
 * @author 梦境迷离
 * @since 2021/6/6
 * @version 1.0
 */
data class BSQLException(val msg: String = "Bitlap SQL Exception", override val cause: Throwable? = null) :
    SQLException(msg, cause)

File: network/src/main/scala/org/bitlap/network/BSQLException.scala
/* Copyright (c) 2022 bitlap.org */
package org.bitlap.network

import java.sql.SQLException

/**
 * @author 梦境迷离
 * @since 2021/11/20
 * @version 1.0
 */
case class BSQLException(
  msg: String = "Bitlap SQL Exception",
  cause: Throwable = null
) extends SQLException(msg, cause)